public/components/navigation.react.js

```diff
@@ -9,12 +9,19 @@ export default class NavigationComponent extends React.Component {
     }
 
     render () {
+        var distinctContentSources = [];
+        var allContentSources = this.props.data;
+        for(var i = allContentSources.length - 1; i >= 0; i--) {
+            if(distinctContentSources[allContentSources[i].id] == undefined)
+                distinctContentSources[allContentSources[i].id] = { appName: allContentSources[i].appName, environment: allContentSources[i].environment };
+        }
```

Comment:
Please note that `=>` [is now available](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)